Adds better visualisation for conversations

TheHuntsman4 opened the pull request.

TheHuntsman4 committed:
fix: fixes the conversations going on forever by limiting number of exchanges per conversation

TheHuntsman4 committed:
feat: updates teh archiving functionality of short term memory

TheHuntsman4 committed:
fix: minor change to the API key name

SreeSharvesh commented:
Observations and To be fixed:

- Output is better looking and organised now
- AK's persona is wrong. It contains the prompt template instead of AK's personality.
- Bingo boards getting updated wrongly. Keyword matching doesn't seem to work and sometimes, when the match is found for one box. a different box gets updated.
- More frequent rate limit warnings in this version.
- Conversation ending to be fixed.

Don't merge it. I'll make a PR with time-independent environment and pls review and merge that PR

SreeSharvesh commented on simulation/agents_personas/AK.txt:
AK's person is incorrect. It's the prompt template

TheHuntsman4 committed:
fix: adds the correct content to AK.txt

TheHuntsman4 committed:
feat: adds safeguards against the experiment crashing due to an indexing issue.
feat: adds methods and to use raw form responses from yamls and updates the prompt temaplate to handle this
